src/hooks/useDirectoryContent.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useSelector } from 'react-redux';
+import React, { useRef, useEffect, useCallback, useState } from 'react';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { fetchElementsInfos } from '../utils/rest-api';
+import { UUID } from 'crypto';
+import { IElement, IElementMetadata, ReduxState } from '../redux/reducer.type';
+
+type useDirectoryContentReturnType = [
+    data: IElement[],
+    childrenMetadata: Record<UUID, IElementMetadata>
+];
+export const useDirectoryContent = (
+    setIsMissingDataAfterDirChange: React.Dispatch<
+        React.SetStateAction<boolean>
+    >
+): useDirectoryContentReturnType => {
+    const currentChildren = useSelector(
+        (state: ReduxState) => state.currentChildren
+    );
+    const [childrenMetadata, setChildrenMetadata] = useState({});
+    const { snackError } = useSnackMessage();
+
+    const [data, setData] = useState(currentChildren);
```

Comment:
rename data to something more specific ? here it's hard to say what we are returning

Reply:
I renamed it to `rows`